### PR TITLE
[ACCEL] Create possibility to generate and download student' grades CSV report

### PR DESCRIPTION
**Pull request by arsentur**
**Description:**
- Create a possibility to generate and download student' grades CSV report
- Add translations for text and button for student' grades report

**Youtrack:**
[ACCEL-458](https://youtrack.raccoongang.com/issue/ACCEL-458)

### CODE REVIEW

File: lms/djangoapps/instructor_task/api.py [hunk 1]
```diff
@@ -14,16 +14,15 @@
...
     check_arguments_for_overriding,
+    check_arguments_for_rescoring,
     check_entrance_exam_problems_for_rescoring,
     encode_entrance_exam_and_student_input,
     encode_problem_and_student_input,
     submit_task
 )
 from lms.djangoapps.instructor_task.models import InstructorTask
 from lms.djangoapps.instructor_task.tasks import (
-    override_problem_score,
```
Comment (flying-pi) [inline lms/djangoapps/instructor_task/api.py]: Nit: why you move it?

File: lms/djangoapps/instructor_task/api.py [hunk 2]
```diff
@@ -471,6 +472,18 @@ def submit_export_ora2_data(request, course_key):
     return submit_task(request, task_type, task_class, course_key, task_input, task_key)
 
 
+def submit_students_grades_data(request, course_key):
+    """
+    AlreadyRunningError is raised if an students grades report is already being generated.
```
Comment (flying-pi) [inline lms/djangoapps/instructor_task/api.py]: Seems title and body of the docstring missing. Also, nice to add some pydoc markup for exception description.

File: lms/djangoapps/instructor_task/tasks.py [hunk 3]
```diff
@@ -40,14 +40,15 @@
     cohort_students_and_upload,
     upload_course_survey_report,
     upload_ora2_data,
-    upload_proctored_exam_results_report
+    upload_proctored_exam_results_report,
+    upload_students_grades_data,
 )
 from lms.djangoapps.instructor_task.tasks_helper.module_state import (
     delete_problem_module_state,
-    perform_module_state_update,
```
Comment (flying-pi) [inline lms/djangoapps/instructor_task/tasks.py]: ditto, why the order is changed?

File: lms/djangoapps/instructor_task/tasks_helper/misc.py [hunk 4]
```diff
@@ -8,15 +8,17 @@
...
 from time import time
 
-import unicodecsv
 from django.contrib.auth.models import User
 from django.core.exceptions import ValidationError
 from django.core.files.storage import DefaultStorage
-from openassessment.data import OraAggregateData
 from pytz import UTC
 
+import unicodecsv
```
Comment (flying-pi) [inline lms/djangoapps/instructor_task/tasks_helper/misc.py]: ditto

File: lms/djangoapps/instructor_task/tasks_helper/misc.py [hunk 5]
```diff
@@ -301,3 +303,126 @@ def upload_ora2_data(
...
+
+    task_progress.update_task_state(extra_meta=curr_step)
+
+    try:
+        for student in students:
+            # Check student access
+            course = get_course_with_access(student, 'load', course_id, check_if_enrolled=True)
+
+            # Get students grade
+            course_grade = CourseGradeFactory().read(student, course)
```
Comment (flying-pi) [inline lms/djangoapps/instructor_task/tasks_helper/misc.py]: Could you move the end of the try block here and set less generic exception, if it possible(i.e. BigGradeException instead Exception)

File: lms/djangoapps/instructor_task/tasks_helper/misc.py [hunk 6]
```diff
@@ -301,3 +303,126 @@ def upload_ora2_data(
     TASK_LOG.info(u'%s, Task type: %s, Upload complete.', task_info_string, action_name)
 
     return UPDATE_STATUS_SUCCEEDED
+
+
+def upload_students_grades_data(
```
Comment (flying-pi) [inline lms/djangoapps/instructor_task/tasks_helper/misc.py]: As for me, will be great, if doc stings will contain schema information

Comment (flying-pi) [reply lms/djangoapps/instructor_task/tasks_helper/misc.py]: Oh, and seems you forget to update it completely

Comment (flying-pi) [inline lms/djangoapps/instructor_task/tasks_helper/misc.py]: Could you move out the logic with grade calculation (gathering grade info, query user, etc) It allows to reduce method size and allows decomposition of the logic.

File: lms/djangoapps/instructor_task/api.py (hunk 1 above)
Comment (arsentur) [reply lms/djangoapps/instructor_task/api.py]: Import order fixed by using `isort` lib.

File: lms/djangoapps/instructor_task/tasks.py (hunk 3 above)
Comment (arsentur) [reply lms/djangoapps/instructor_task/tasks.py]: Import order fixed by using `isort` lib.

File: lms/djangoapps/instructor_task/tasks_helper/misc.py (hunk 4 above)
Comment (arsentur) [reply lms/djangoapps/instructor_task/tasks_helper/misc.py]: Import order fixed by using `isort` lib.

File: lms/djangoapps/instructor_task/tasks_helper/misc.py (hunk 6 above)
Comment (arsentur) [reply lms/djangoapps/instructor_task/tasks_helper/misc.py]: Util function that generates students grades contain schema information in the docstring

Comment (arsentur) [reply lms/djangoapps/instructor_task/tasks_helper/misc.py]: Done

File: lms/djangoapps/instructor_task/api.py (hunk 2 above)
Comment (arsentur) [reply lms/djangoapps/instructor_task/api.py]: I used a template solution. And, I think, it is not necessary for this function.

**Pull request merged**